src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -658,7 +658,7 @@ internal static string ParseSolutionFromSolutionFilter(string solutionFilterFile
                 JsonDocumentOptions options = new JsonDocumentOptions() { AllowTrailingCommas = true, CommentHandling = JsonCommentHandling.Skip };
                 JsonDocument text = JsonDocument.Parse(File.ReadAllText(solutionFilterFile), options);
                 solution = text.RootElement.GetProperty("solution");
-                return FileUtilities.GetFullPath(solution.GetProperty("path").GetString(), Path.GetDirectoryName(solutionFilterFile));
+                return FileUtilities.GetFullPath(solution.GetProperty("path").GetString(), Path.GetDirectoryName(solutionFilterFile), escape: false);
```

Comment:
Could this be

```suggestion
                return FileUtilities.NormalizePath(Path.GetDirectoryName(solutionFilterFile), solution.GetProperty("path").GetString());
```

without changing `GetFullPath`?

Reply:
no, this won't work because in  GetFullPath before normalization the solution path that is provided in slnf needs to be fixed for the current machine. 

https://github.com/dotnet/msbuild/blob/04b6e1b59df1004aeedc4703757a5c407c06c139/src/Shared/FileUtilities.cs#L759

I can create new function GetFullPathUnescaped and duplicate the code there. What do you think?

Reply:
is it possible to do here

`string fullPath = NormalizePath(Path.Combine(currentDirectory, FileUtilities.FixFilePath(EscapingUtilities.UnescapeAll(fileSpec)));`
 instead of making modification of `FileUtilities.cs`?

Reply:
there is also adding trailing slash at the end of the full path
https://github.com/dotnet/msbuild/blob/aff54559404d31214c71aa2ea6d2caa6003b0334/src/Shared/FileUtilities.cs#L764-L772

Not modifying the `FileUtilities.GetFullPath` by adding extra check makes sense because it is on a hot path. Adding a new method `FileUtilities.GetFullPathUnescaped` that will duplicate most of the code except escaping would be a compromise for that reason.

Is there other reason not to modify `FileUtilities.cs` that I miss?

Reply:
let's add more info on when path can be not escaped and it looks ok for me.

Reply:
thanks! added explanatory comments for not escaping